### ISSUE TITLE
[Build] OSX arm64 on main eigen pset1 candidate function template not viable

### ISSUE THREAD

**Issue by devigned** [microsoft/onnxruntime#18322]
### Describe the issue

After the recent changes to update the eigen sha for deps.txt, I have been unable to build running `./build.sh` on the main branch.

Previous to the changes to eigen, I was able to build on OSX arm64. 

related #18286 

### Urgency

This is probably not urgent, but it will likely affect others with the same OS / Arch.

### Target platform

darwin/arm64

### Build script

`./build.sh --parallel` or any other variation of `./build.sh`

### Error / output

```
[ 71%] Building CXX object CMakeFiles/onnxruntime_providers.dir/Users/david/code/msft/onnxruntime/onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc.o
[ 71%] Building CXX object CMakeFiles/onnxruntime_providers.dir/Users/david/code/msft/onnxruntime/onnxruntime/contrib_ops/cpu/quantization/dynamic_quantize_lstm.cc.o
[ 71%] Building CXX object CMakeFiles/onnxruntime_providers.dir/Users/david/code/msft/onnxruntime/onnxruntime/contrib_ops/cpu/quantization/dynamic_quantize_matmul.cc.o
In file included from /Users/david/code/msft/onnxruntime/onnxruntime/contrib_ops/cpu/inverse.cc:8:
In file included from /Users/david/code/msft/onnxruntime/onnxruntime/core/util/math_cpuonly.h:68:
In file included from /Users/david/code/msft/onnxruntime/build/MacOS/Debug/_deps/eigen-src/Eigen/Core:358:
/Users/david/code/msft/onnxruntime/build/MacOS/Debug/_deps/eigen-src/Eigen/src/Core/PartialReduxEvaluator.h:57:64: error: no matching function for call to 'pset1'
PacketType packetwise_redux_empty_value(const Func& ) { return pset1<PacketType>(0); }
```

### Visual Studio Version

N/A

### GCC / Compiler Version

$ clang -v Apple clang version 15.0.0 (clang-1500.0.40.1) Target: arm64-apple-darwin23.1.0 Thread model: posix InstalledDir: /Library/Developer/CommandLineTools/usr/bin

**Comment by yufenglee** [comment]
This PR should fix the issue: https://github.com/microsoft/onnxruntime/pull/18308

**Comment by devigned** [comment]
> This PR should fix the issue: #18308

That seems like it's pointed at 1.16.2 rather than main. #18200 fixed the problem for me when I compiled and tested locally on darwin/arm64.
